web3/utils/transactions.py

```diff
@@ -168,3 +170,28 @@ def get_buffered_gas_estimate(web3, transaction, gas_buffer=100000):
         )
 
     return min(gas_limit, gas_estimate + gas_buffer)
+
+
+def prepare_replacement_transaction(web3, current_transaction, new_transaction):
+    if not current_transaction:
+        raise ValueError('Supplied transaction with hash {} does not exist'
+                         .format(current_transaction['hash']))
+    if current_transaction['blockHash'] is not None:
+        raise ValueError('Supplied transaction with hash {} has already been mined'
+                         .format(current_transaction['hash']))
+    if 'nonce' in new_transaction and new_transaction['nonce'] != current_transaction['nonce']:
+        raise ValueError('Supplied nonce in new_transaction must match the pending transaction')
+
+    if 'nonce' not in new_transaction:
+        new_transaction = assoc(new_transaction, 'nonce', current_transaction['nonce'])
+
+    # TODO: Possibly in a middlware somehow?
```

Comment:
Not sure this can go in the middleware, because all the middleware can see is the method being called: `eth_sendTransaction`. That's not enough to know that we should use this special replacement gas price approach.